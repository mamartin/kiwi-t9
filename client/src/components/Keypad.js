// @flow
import React from "react"
import { View } from "react-native"

// components
import { Button } from "."

const styles = {
  keypadWrapper: {
    flexDirection: "row",
    flexWrap: "wrap",
    justifyContent: "center",
  },
}

type KeypadProps = {|
  onPress: string => null,
|}

const keypad = [
  { number: "1", letters: "" },
  { number: "2", letters: "abc" },
  { number: "3", letters: "def" },
  { number: "4", letters: "ghi" },
  { number: "5", letters: "jkl" },
  { number: "6", letters: "mno" },
  { number: "7", letters: "pqrs" },
  { number: "8", letters: "tuv" },
  { number: "9", letters: "vwxy" },
  { number: "*", letters: "DELETE" },
  { number: "0", letters: "REAL" },
  { number: "#", letters: "RESET" },
]

export default class Keypad extends React.PureComponent<KeypadProps> {
  render() {
    const { onPress } = this.props
    return (
      <View style={styles.keypadWrapper}>
        {keypad.map(button => (
          <Button
            key={button.number}
            onPress={() => onPress(button.number)}
            number={button.number}
          >
            {button.letters}
          </Button>
        ))}
      </View>
    )
  }
}
