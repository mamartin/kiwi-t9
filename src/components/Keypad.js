// @flow
import React from "react"
import { View } from "react-native"

// components
import { Button } from "."

const styles = {
  keypadWrapper: {
    flexDirection: "row",
    flexWrap: "wrap",
  },
}

type KeypadProps = {|
  onChange: string => null,
|}

type KeypadState = {
  numbers: string,
}

const keypad = [
  { number: "1", letters: "" },
  { number: "2", letters: "abc" },
  { number: "3", letters: "def" },
  { number: "4", letters: "ghi" },
  { number: "5", letters: "jkl" },
  { number: "6", letters: "mno" },
  { number: "7", letters: "pqrs" },
  { number: "8", letters: "tuv" },
  { number: "9", letters: "vwxy" },
]

export default class Keypad extends React.PureComponent<
  KeypadProps,
  KeypadState,
> {
  state = {
    numbers: "",
  }

  handleButtonPressed = (number: string) => {
    const { onChange } = this.props
    const { numbers } = this.state

    this.setState({ numbers: `${numbers}${number}` }, () => {
      const { numbers } = this.state
      onChange(numbers)
    })
  }

  render() {
    // const { onPress } = this.props
    return (
      <View style={styles.keypadWrapper}>
        {keypad.map(button => (
          <Button
            key={button.number}
            onPress={() => this.handleButtonPressed(button.number)}
          >
            {button.letters}
          </Button>
        ))}
      </View>
    )
  }
}
