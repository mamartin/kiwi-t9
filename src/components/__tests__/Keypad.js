import React from "react"
import renderer from "react-test-renderer"

// components
import Keypad from "../Keypad"

describe("Keypad component", () => {
  it("renders with default props", () => {
    const tree = renderer.create(<Keypad onChange={() => null} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
