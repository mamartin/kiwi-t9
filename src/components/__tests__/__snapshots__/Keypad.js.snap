// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad component renders with default props 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      abc
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      def
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      ghi
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      jkl
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      mno
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      pqrs
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      tuv
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "grey",
        "borderRadius": 60,
        "height": 60,
        "justifyContent": "center",
        "opacity": 1,
        "width": "33%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      vwxy
    </Text>
  </View>
</View>
`;
